osal/freertos/osal_common.h

```diff
@@ -84,4 +84,10 @@ osal_sighandler_t osal_signal(int signum, osal_sighandler_t handler);
 int osal_sigprocmask(int how, const sigset_t *set, sigset_t *oldset);
 int osal_sigemptyset(sigset_t *set);
 int osal_sigaddset(sigset_t *set, int signum);
+
```

Comment:
Inline to the ongoing discussion regarding osal file/dir structuring... 

- xxx/osal_common.h files currently seem redundant
- We could have one `osal/common.h` common for all target platforms, having common includes/types/declarations/abstractions.
- All platform specific includes/types/declarations in `osal/linux/osal.h` and `osal/freertos/osal.h`
- All platform specific definitions in `osal/linux/osal.c` and `osal/freertos/osal.c`

```
└── osal
    ├── common.h
    ├── freertos
    │   ├── osal.c
    │   └── osal.h
    └── linux
        ├── osal.c
        └── osal.h
```
**Does this sound good?**

Reply:
I would like to suggest doing that change in another PR. The goal of that one is to add the new allocator APIs.

Reply:
Re: the signal related functions.  Does FreeRTOS have a similar signal mechanism?  Is the Linux implementation the only signal user, and is it just Linux timer related functionality?  If the answers are NO, YES, and YES ... IMO abstract signal away by adding start/stop/create/delete timer() or equivalent to osal_common.h (with target specific implementations).

Reply:
We don't use signals or signal like implementation (event flags) in freertos port. The alarm signal is only used for linux target to have timer implementation using the linux kernel features. I have already created a simple solution and using freertos timer thread instead. I note that, the PR doesn't cover this topic, it's only a simple osal refactor to make osal portable for embedded platform.

Reply:
Suggestion.   Abstract signal away by adding start/stop/create/delete timer() or equivalent to osal_common.h (with target specific implementations).